### PR TITLE
Bug 1189902 - Prompt user asking to report crash if they have not opted in/out

### PR DESCRIPTION
**Pull request by sleroux**

### CODE REVIEW

File: Client/Extensions/UIAlertControllerExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    - parameter sendReportCallback: Send report option handler
+    - parameter alwaysSendCallback: Always send option handler
+    - parameter dontSendCallback:   Dont send option handler
+    - parameter neverSendCallback:  Never send option handler
+
+    - returns: UIAlertController for opting into crash reporting after a crash occurred
+    */
+    class func crashOptInAlert(
+        sendReportCallback sendReportCallback: UIAlertActionCallback,
```
Comment (rnewman) [inline Client/Extensions/UIAlertControllerExtensions.swift]: This doesn't need a variable name addition — they're the same!

File: Client/Extensions/UIAlertControllerExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        dontSendCallback: UIAlertActionCallback) -> UIAlertController {
+
+        let alert = UIAlertController(
+            title: NSLocalizedString("Oops! Firefox crashed", comment: "Title for prompt displayed to user after the app crashes"),
+            message: NSLocalizedString("Send a crash report so Mozilla can fix the problem?", comment: "Message displayed in the crash dialog above the buttons used to select when sending reports"),
+            preferredStyle: UIAlertControllerStyle.Alert
+        )
+
+        let sendReport = UIAlertAction(
+            title: NSLocalizedString("Send Report", comment: "Used as a button label for crash dialog prompt"),
```
Comment (rnewman) [inline Client/Extensions/UIAlertControllerExtensions.swift]: I think all of your added strings should have named tables. Ask @st3fan what the current style is for doing that.

File: Client/Extensions/UIAlertControllerExtensions.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        )
+
+        let noOption = UIAlertAction(
+            title: NSLocalizedString("No", comment: "Restore Tabs Negative Action"),
+            style: UIAlertActionStyle.Cancel,
+            handler: noCallback
+        )
+
+        let okayOption = UIAlertAction(
+            title: NSLocalizedString("Okay", comment: "Restore Tabs Affirmative Action"),
```
Comment (rnewman) [inline Client/Extensions/UIAlertControllerExtensions.swift]: Isn't iOS style to make this the affirmative verb: "Restore"? @tecgirl?

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 4]
```diff
@@ -432,6 +431,47 @@ class BrowserViewController: UIViewController {
...
+        func addAndSelect() {
+            let tab = tabManager.addTab()
+            tabManager.selectTab(tab)
+        }
+
+        let alert = UIAlertController.restoreTabsAlert(
+            okayCallback: { _ in
+                self.restoreTabs()
+                if self.tabManager.count == 0 {
+                    addAndSelect()
```
Comment (rnewman) [inline Client/Frontend/Browser/BrowserViewController.swift]: All of this logic seems like it should be in `TabManager`.

Comment (sleroux) [inline Client/Frontend/Browser/BrowserViewController.swift]: I've moved the restore tabs logic we have in the BVC to the restoreTabs method in tabManager

File: Client/Extensions/UIAlertControllerExtensions.swift (hunk 1 above)
Comment (sleroux) [inline Client/Extensions/UIAlertControllerExtensions.swift]: I wanted to keep the first param to be explicit when calling the function so it's clear to the external caller that we are assigning the callback to the sendReport action. I know in Swift < 2.0 I could use the # sign to force the explicit first param but as far as I know in Swift 2.0 you just have to duplicate it :(

**Pull request merged**